core/src/main/scala/kafka/server/BrokerBlockingSender.scala

```diff
@@ -148,8 +148,15 @@ class RemoteBrokerBlockingSender(sourceBroker: BrokerEndPoint,
 
     val channelBuilder = ClientUtils.createChannelBuilder(brokerConfig, time, logContext)
     val selector = new Selector(
+      NetworkReceive.UNLIMITED,
       brokerConfig.getLong(CommonClientConfigs.CONNECTIONS_MAX_IDLE_MS_CONFIG),
-      metrics, time, "remote-readonly-" + clientId, channelBuilder, logContext)
+      metrics,
+      time,
+      "remote-readonly-" + clientId,
+      Map("broker-id" -> sourceBroker.id.toString, "fetcher-id" -> fetcherId.toString).asJava,
```

Comment:
So the tags are:
`"leader-broker-id" + "fetcher-id"`
Thanks for finding out the root cause.

Question: So, in the source cluster, the leader changes from 2 -> 3, it works now because it has different metrics tags. But what about the leader change from 3 -> 2 next? Will the metrics duplicate with a previously created one? Should we remove the metrics or something when remove fetcher? How does "normal" fetcher do on this?

Reply:
Thanks for the explanation, Fede!